Clean up nbdev GitHub

Kamyarmb opened the pull request.

Kamyarmb committed:
test_push

Kamyarmb committed:
deleting the files related to nbdev

Kamyarmb committed:
deleting the files related to nbdev

Kamyarmb committed:
deleting the files related to nbdev

Kamyarmb committed:
deleting the files related to nbdev

Kamyarmb committed:
deleting the files related to nbdev

Kamyarmb committed:
deleting the files related to nbdev

Kamyarmb committed:
deleting the files related to nbdev

Kamyarmb committed:
deleting the files related to nbdev

Kamyarmb committed:
deleting the files related to nbdev

Kamyarmb committed:
deleting the files related to nbdev

Kamyarmb committed:
deleting the files related to nbdev

Kamyarmb committed:
deleting the files related to nbdev

Kamyarmb committed:
deleting the files related to nbdev

Kamyarmb committed:
deleting the files related to nbdev

Kamyarmb committed:
deleting the files related to nbdev

Kamyarmb committed:
deleting the files related to nbdev

Kamyarmb committed:
deleting the files related to nbdev

Kamyarmb committed:
deleting the files related to nbdev

Kamyarmb committed:
deleting the files related to nbdev

Kamyarmb committed:
deleting the files related to nbdev

Kamyarmb committed:
deleting the files related to nbdev

Kamyarmb committed:
deleting the files related to nbdev

Kamyarmb committed:
deleting the files related to nbdev

paulinebellingmo committed:
check push

Kamyarmb committed:
check

kevinksyTRD commented:
We shouldn't delete all the notebooks as they provide a lot of information on the library, equations, sources, etc. Keeping the notebook doesn't mean that we have to use nbdev framework. I recommend to undo all the deletion of the notebooks.

kevinksyTRD commented on ship_model_lib/ship_model_lib/added_resistance.py:
```diff
@@ -35,7 +12,7 @@
 from scipy.special import gamma
 
-from .types import WaveSpectrumType, ShipType
+from .ship_types import WaveSpectrumType, ShipType
```
Was this change necessary? This may make the library backward incompatible.

Kamyarmb replied:
It was Pytest could not run it, I have updated the pytest, all of the functions test have been transferred to Pytest

kevinksyTRD commented on ship_model_lib/ship_model_lib/calm_water_resistance.py:
```diff
@@ -28,7 +7,7 @@
 import numpy as np
 from .utility import get_interpolation_1d_function, Interpolated1DValue
-from .types import ResistanceLevel
+from .ship_types import ResistanceLevel
```
The same comment as above.

kevinksyTRD commented on ship_model_lib/ship_model_lib/data/create_table.py:
```diff
@@ -3,7 +3,7 @@
 from scipy import interpolate
 
-from ship_model_lib.types import ShipType
+from ship_model_lib.ship_types import ShipType
```
The same comment as above.

kevinksyTRD commented on ship_model_lib/settings.ini:
This config file is a part of nbdev framework. It would be better to use more common config files for building such as pyproject.toml. This toml file should replace both settings.ini and setup.py files.

Kamyarmb replied:
I applied it.

Kamyarmb replied:
nbdev related stuff have been removed from this file, is it enough?

Kamyarmb committed:
check

Commit:
Fixed some failing tests

Commit:
back to ship_types

Commit:
fix: pending check by restoring workflow name

Commit:
feat: migrate to uv build system and update dependencies

- Migrate from pip/setuptools to uv for faster builds
- Add ruff for linting and formatting
- Update dependency version constraints in pyproject.toml
- Modernize package structure and configuration
- Update GitHub workflows to use uv and ruff
- Fix package data inclusion (dependencies/ moved into package)
- Add comprehensive migration documentation

This migration provides 10-100x faster dependency installation
and improves build reproducibility with uv.lock.

Commit:
fix: ruff check cleared

Commit:
fix: failing tests

Commit:
fix: renderer option removed

Commit:
fix: name for the workflow reverted

Commit:
fix: test yml name reverted

Commit:
chore: trigger status refresh
